android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java

```diff
@@ -243,56 +256,67 @@ public MatchingEngineHelper(final Builder builder) {
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyTestTriggerMode);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyUpdateInterval);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyMaxNumUpdates);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyTestPort);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyThreshold);
+        onEdgeEventPreferenceChanged(prefs, prefKeyPerfMarginSwitch);
+        onEdgeEventPreferenceChanged(prefs, prefKeyAutoMigration);
     }
 
-    public void startEdgeEvents() {
+    public void startEdgeEvents(boolean newConnection) {
+        if (!mEdgeEventsEnabled) {
+            Log.e(TAG, "Aborting attempt to startEdgeEvents while mEdgeEventsEnabled="+ false);
+            return;
+        }
         if (mClosestCloudlet == null) {
             Log.w(TAG, "Must perform findCloudlet before starting Edge Events.");
             return;
         }
-        mEdgeEventsConfig = me.createDefaultEdgeEventsConfig();
-        Log.i(TAG, "Default EdgeEventsConfig="+mEdgeEventsConfig);
-
-        if (mOverrideDefaultConfig) {
-            // In this case, all attributes for mEdgeEventsConfig will be set from Preferences.
-            // Otherwise, we retain the default values.
-            initEdgeEventsConfigPrefs();
-            Log.i(TAG, "Custom EdgeEventsConfig="+mEdgeEventsConfig);
-        }
 
         new Thread(() -> {
+            mEdgeEventsConfig = me.createDefaultEdgeEventsConfig();
+            String message = "Using Default EdgeEventsConfig=" + mEdgeEventsConfig;
+            Log.i(TAG, message);
+
+            if (mOverrideDefaultConfig) {
+                // In this case, all attributes for mEdgeEventsConfig will be set from Preferences.
+                // Otherwise, we retain the default values.
+                initEdgeEventsConfigPrefs();
+                message = "Using Custom EdgeEventsConfig="+mEdgeEventsConfig;
+                Log.i(TAG, message);
+            }
+            meHelperInterface.showMessage(message);
+
             // Only do this once.
             if (mEdgeEventsSubscriber == null) {
                 mEdgeEventsSubscriber = new EdgeEventsSubscriber();
                 me.getEdgeEventsBus().register(mEdgeEventsSubscriber);
             }
 
-            Log.i(TAG, "mEdgeEventsConfig="+mEdgeEventsConfig.toString());
-            meHelperInterface.showMessage("mEdgeEventsConfig="+mEdgeEventsConfig.toString());
-
-            String message;
-            if (me.isAutoMigrateEdgeEventsConnection()) {
-                message = "Auto-migrating to new app inst";
-            } else {
-                if (mEdgeEventsRunning) {
+            if (mEdgeEventsRunning) {
+                if (me.isAutoMigrateEdgeEventsConnection()) {
+                    if (newConnection) {
+                        message = "Retaining connection to existing app inst";
+                    } else {
+                        message = "Auto-migrating to new app inst";

```

Comment:
I don't know really  understand the naming of newConnection here.
```
startEdgeEvents(newConnection: true) -> AutoMigrate is true -> "Retaining connection to existing app inst"
startEdgeEvents(newConnection: false) -> AutoMigrate is true -> "Auto-migrating to new app inst"
```
If this is the behavior  shall it be renamed to sameConnection or oldConnection, or I am missing something ?

Reply:
You are correct that this doesn't make sense. Will reverse the messages. Thanks for catching this.